scripts/ensemble_kalman_paw_example.py

```diff
@@ -0,0 +1,251 @@
+for i in range(len(left_cam_ensemble_preds)):
+    left_paw_ensemble_preds[i] = np.concatenate((left_cam_ensemble_preds[i][:2], right_cam_ensemble_preds[i][:2]))
+    right_paw_ensemble_preds[i] = np.concatenate((left_cam_ensemble_preds[i][2:4], right_cam_ensemble_preds[i][2:4]))
+    
+left_paw_ensemble_vars = np.zeros((left_cam_ensemble_vars.shape[0], left_cam_ensemble_vars.shape[1]))
+right_paw_ensemble_vars = np.zeros((right_cam_ensemble_vars.shape[0], right_cam_ensemble_vars.shape[1]))
+for i in range(len(left_cam_ensemble_vars)):
+    left_paw_ensemble_vars[i] = np.concatenate((left_cam_ensemble_vars[i][:2], right_cam_ensemble_vars[i][:2]))
+    right_paw_ensemble_vars[i] = np.concatenate((left_cam_ensemble_vars[i][2:4], right_cam_ensemble_vars[i][2:4]))
+    
+#get mean of each paw
+mean_camera_l_x = good_stacked_ensemble_preds[:,0].mean()
+mean_camera_l_y = good_stacked_ensemble_preds[:,1].mean()
+mean_camera_r_x = good_stacked_ensemble_preds[:,2].mean()
+mean_camera_r_y = good_stacked_ensemble_preds[:,3].mean()
+means_camera = [mean_camera_l_x, mean_camera_l_y, mean_camera_r_x, mean_camera_r_y]
+
+left_paw_ensemble_stacks = np.concatenate((left_cam_ensemble_stacks[:,:,:2], right_cam_ensemble_stacks[:,:,:2]),2)
+scaled_left_paw_ensemble_stacks = remove_camera_means(left_paw_ensemble_stacks, means_camera)
+
+right_paw_ensemble_stacks = np.concatenate((right_cam_ensemble_stacks[:,:,:2], right_cam_ensemble_stacks[:,:,:2]),2)
+scaled_right_paw_ensemble_stacks = remove_camera_means(right_paw_ensemble_stacks, means_camera)
+
+good_scaled_stacked_ensemble_preds = remove_camera_means(good_stacked_ensemble_preds[None,:,:], means_camera)[0]
+ensemble_pca, ensemble_ex_var = pca(good_scaled_stacked_ensemble_preds, 3)
+
+scaled_left_paw_ensemble_preds = remove_camera_means(left_paw_ensemble_preds[None,:,:], means_camera)[0]
+ensemble_pcs_left_paw = ensemble_pca.transform(scaled_left_paw_ensemble_preds)
+good_ensemble_pcs_left_paw = ensemble_pcs_left_paw[good_frames]
+
+scaled_right_paw_ensemble_preds = remove_camera_means(right_paw_ensemble_preds[None,:,:], means_camera)[0]
+ensemble_pcs_right_paw = ensemble_pca.transform(scaled_right_paw_ensemble_preds)
+good_ensemble_pcs_right_paw = ensemble_pcs_right_paw[good_frames]
+
+# # Kalman Filtering + Smoothing
+# $z_t = (d_t, x_t, y_t)$
+# $z_t = A z_{t-1} + e_t, e_t ~ N(0,E)$
+# $O_t = B z_t + n_t, n_t ~ N(0,D_t)$
+
+# ## Set parameters
+if paw == 'left':
+    save_keypoint_name = keypoint_names[0]
+    good_ensemble_pcs = good_ensemble_pcs_left_paw
+    ensemble_vars = left_paw_ensemble_vars
+    y = scaled_left_paw_ensemble_preds
+    ensemble_stacks = scaled_left_paw_ensemble_stacks
+else:
+    save_keypoint_name = keypoint_names[1]
+    good_ensemble_pcs = good_ensemble_pcs_right_paw
+    ensemble_vars = right_paw_ensemble_vars
+    y = scaled_right_paw_ensemble_preds
+    ensemble_stacks = scaled_right_paw_ensemble_stacks
```

Comment:
instead of doing if/else here would be better to do a for loop over paws so we get both automatically when running the scripts